File: library/src/main/java/cn/wandersnail/widget/dialog/DialogEventObserver.java
package cn.wandersnail.widget.dialog;

import android.os.Bundle;

/**
 * date: 2019/8/22 14:10
 * author: zengfansheng
 */
public interface DialogEventObserver {
    default void onCreate(Bundle savedInstanceState) {
    }

    default void onStart() {
    }

    default void onStop() {
    }

    default void onShow() {
    }

    default void onDismiss() {
    }

    default void onCancel() {
    }

    default void onBackPressed() {
    }

    default void onAttachedToWindow() {
    }

    default void onDetachedFromWindow() {
    }

    default void onContentChanged() {
    }

    default void onWindowFocusChanged(boolean hasFocus) {
    }
}

File: library/src/main/java/com/snail/widget/dialog/DialogEventObserver.kt
package com.snail.widget.dialog

import android.os.Bundle

/**
 *
 *
 * date: 2019/5/6 14:26
 * author: zengfansheng
 */
interface DialogEventObserver {
    fun onCreate(savedInstanceState: Bundle?)

    fun onStart()

    fun onStop()

    fun onShow()

    fun onDismiss()

    fun onCancel()

    fun onBackPressed()

    fun onAttachedToWindow()

    fun onDetachedFromWindow()

    fun onContentChanged()

    fun onWindowFocusChanged(hasFocus: Boolean)
}